File: go/two-fer/two_fer.go
// Package twofer is an exercism solution
package twofer

import (
	"fmt"
)

// ShareWith is a string utility which returns
// a string containing "One for {name}, one for me"
func ShareWith(name string) string {
	if name == "" {
		name = "you"
	}
	return fmt.Sprintf("One for %s, one for me.", name)
}

File: csharp/two-fer/TwoFer.cs
using System;

public static class TwoFer
{
    public static string Speak(String name = null) =>
        $"One for {Default(name)}, one for me.";

    private static string Default(String name = null) =>
        string.IsNullOrEmpty(name) ? "you" : name;

}
